lib/ts/stacks.ts

```diff
@@ -76,7 +76,66 @@ namespace Stacks {
 
         return Controller;
     }
+
     export function addController(name: string, controller: ControllerDefinition) {
         application.register(name, createController(controller));
-    };
+    }
+
+    export interface TooltipOptions {
+        placement: string;
+        classNames: string;
+    }
+
+    export function setTooltipHtml(element: Element, html: string, options?: TooltipOptions) {
+        setTooltip(element, popover => popover.innerHTML = html, options);
+    }
+
+    export function setTooltipText(element: Element, text: string, options?: TooltipOptions) {
+        setTooltip(element, popover => popover.textContent = text, options);
+    }
+
+    var generatedTooltipIdNumber = 0;
+
+    function setTooltip(element: Element, fn: (popover: HTMLElement) => void, options?: TooltipOptions) {
```

Comment:
Please add jsdocs here. I have no idea what the use of `fn` is without looking at calling code.